### PR TITLE
Refine some datanode code

### PR DESCRIPTION
**Pull request by SimFG**
#18352 
Signed-off-by: SimFG <bang.fu@zilliz.com>

### CODE REVIEW

File: internal/datanode/data_node.go [hunk 1]
```diff
@@ -457,9 +457,9 @@ func (node *DataNode) Start() error {
 		},
 		Count: 1,
 	})
-	if err != nil {
-		log.Warn("fail to alloc timestamp", zap.Error(err))
-		return err
+	if rep.Status.ErrorCode != commonpb.ErrorCode_Success || err != nil {
```
Comment (czs007) [inline internal/datanode/data_node.go:460]: test err != nil first.
rep may be nil and access Status casue program panic.

File: internal/datanode/meta_service.go [hunk 2]
```diff
@@ -49,28 +49,11 @@ func newMetaService(rc types.RootCoord, collectionID UniqueID) *metaService {
...
-			MsgID:     0, //GOOSE TODO
-			Timestamp: 0, // GOOSE TODO
-			SourceID:  Params.DataNodeCfg.GetNodeID(),
-		},
-		DbName:       "default", // GOOSE TODO
-		CollectionID: collID,
-		TimeStamp:    timestamp,
+	response, err := mService.getCollectionInfo(ctx, collID, timestamp)
+	if response != nil {
+		return response.GetSchema(), err
```
Comment (czs007) [inline internal/datanode/meta_service.go:54]: why remove the check of ErrorCode()?

Comment (SimFG) [reply internal/datanode/meta_service.go:54]: Before modification, `getCollectionSchema` and `getCollectionInfo` were the same. The check of ErrorCode() has been included in the `getCollectionInfo`.

File: internal/datanode/data_node.go (hunk 1 above)
Comment (SimFG) [reply internal/datanode/data_node.go]: Has updated.

File: internal/datanode/segment_replica.go [hunk 3]
```diff
@@ -298,14 +298,7 @@ func (replica *SegmentReplica) listCompactedSegmentIDs() map[UniqueID][]UniqueID
...
 	for segID, seg := range replica.compactedSegments {
-		var from []UniqueID
-		from, ok := compactedTo2From[seg.compactedTo]
-		if !ok {
-			from = []UniqueID{}
-		}
-
-		from = append(from, segID)
-		compactedTo2From[seg.compactedTo] = from
+		compactedTo2From[seg.compactedTo] = append(compactedTo2From[seg.compactedTo], segID)
```
Comment (czs007) [inline internal/datanode/segment_replica.go:301]: compactedTo2From[seg.compactedTo] may get nil, if seg.compactedTo not existed in compactedTo2From.
Is append nil  right ?

**Pull request merged**